### ISSUE TITLE
Customization Options

### ISSUE THREAD

**Issue by Polgara01** [ivarbrandt/skin.fentastic#23]
I've tried a few skins but they all stray too far from the default Kodi experience, so I've never kept one installed past the initial testing. After a decade of using Kodi, I'd like to thank you for building and maintaining this skin and finally providing an option that works for our setup.  

If anything I suggest is feasible, but you don't have time to implement, I'd be happy to help if you point me in the right direction.  I've forked an addon in the past for personal use, but I've never worked on a skin. 

Now for some unsolicited feedback:

General:
Thank you for the latest update.  Having more on each screen is great.  I would be careful at reducing things any further as it could become difficult for users with vision problems and/or the elderly.  

I saw that you played around with the idea of removing the InfoPanel.  I hope this never happens, or that if it is ever implemented that it would be an option.  This would stray too far from the default Kodi experience for my setup.

Is it possible to customize the subordinate pages inside Fen to make the user experience a bit more consistent?  For instance, I have "Trending Recent" as a widget.  When I use the next page arrow it takes me into default Fen.  I would like to have your Poster/InfoPanel setup on these pages as an option.  I also use a Category widget to access the various sections within Fen 'Movies' and 'TV Shows'.  Poster works great for sections that lead straight to content, such as 'Trending', etc. but LandscapeInfo works better for sections like 'Genre' or 'Year' which otherwise (as Poster/Landscape) show the same icon with no text for every item.  Would it be possible to configure by section if this was implemented?

Stacked Widgets:
I like the stacked widgets, but I don't like the default behavior.  Autoloading the subordinate widget creates a 'rushed' experience for me.  I like to casually scroll through the options and not have it autoload,  I haven't looked at the code and I don't know what is possible, but would you consider an option to have the autoloading of sub-widgets optional so that we can click to load the item?

Scrolling Text:
I see that you added scrolling to the LandscapeInfo view.  This creates an issue on my setup as every visible item will scroll at the same time when I first enter the screen, which I find distracting.  Can you: 1) Make the scrolling optional; or 2) Only scroll the highlighted item?

Live TV:
Is it possible to add customization options to the Live TV section?  For instance: We would like the option to right click and remove items from the 'Recently played channels' section.  We have a shared TV and would like to customize this so we don't see everything everyone else has watched or to remove something we may have clicked on accidentally.  

We would also like to add widgets for things like our favorite lists or custom groups.  

We use IPTV Merge.  I'm not sure if any of the items/sections we see come from this addon rather than from default Kodi, but it would be nice to be able to customize.

In summary: 
Thank you again for the work you've done on this skin!

**Comment by ivarbrandt** [comment]
Firstly, let me express my gratitude for your kind words and appreciation for the Kodi skin I've built and maintained. It's truly rewarding to know that it has finally provided a satisfactory option for your setup after a decade of using Kodi.

> If anything I suggest is feasible, but you don't have time to implement, I'd be happy to help if you point me in the right direction.

It's not so much about the time to implement for me. It's about what I want and don't want in the skin. My goal when I created the skin was for it to be fast, efficient, and have some more advanced features. Adding a lot of optional stuff would defeat this purpose as it would bloat the skin. 

Now, moving on to your feedback.

>  Having more on each screen is great. I would be careful at reducing things any further as it could become difficult for users with vision problems and/or the elderly.

I completely agree. I myself was shocked at the difference when I changed the size of the widgets. But in about a week my eyes adjusted and now I really like the changes. No worries about the size changing anytime soon.

> I saw that you played around with the idea of removing the InfoPanel.

I did think of doing this for a little while. This was because I wanted to fit more items from the widgets on one page. Getting rid of the info panel would have allowed me to set up 6 or 7 poster items per row. But I realized that I would have to go into Fen extras in order to see the plot and other info every time. This would be cumbersome and not very user friendly so decided against the removal of the InfoPanel and instead opted to resize the widgets to what you see on the latest version today.

> When I use the next page arrow it takes me into default Fen. I would like to have your Poster/InfoPanel setup on these pages as an option

I'm not too sure what you mean by this. You can set the View to either InfoWall or WideInfoWall inside the Fen addon in order to have the same InfoPanel from the widgets.

> I also use a Category widget to access the various sections within Fen 'Movies' and 'TV Shows'. Poster works great for sections that lead straight to content, such as 'Trending', etc. but LandscapeInfo works better for sections like 'Genre' or 'Year' which otherwise (as Poster/Landscape) show the same icon with no text for every item. Would it be possible to configure by section if this was implemented?

What do you mean 'configure by section'? And are you referring to the the category widget inside a stacked widget? If you are then no, there's no way that I know of to dynamically change the type of widget that displays once it is set up. I will look into this though as it could be a useful feature. Something that you can do as a workaround is make Shortcut folders within Fen and point the category widget to those Shortcut folders. This way you can choose every single item that shows up in the category widget. 

>  Autoloading the subordinate widget creates a 'rushed' experience for me. I like to casually scroll through the options and not have it autoload, I haven't looked at the code and I don't know what is possible, but would you consider an option to have the autoloading of sub-widgets optional so that we can click to load the item?

I understand where you're coming from regarding this. I too didn't like that the widget loads right away as soon as it was focused. So I implemented a delay in loading the widget after it is focused. This way you can quickly browse the category widget without loading unnecessary lists in between. 

What you are proposing though poses a problem because clicking the item acts as a shortcut to the Fen addon. This way you can load way more content than the widget has. Also, to be quite honest, this request defeats the entire purpose of the stacked widgets. If you want to wait to click to load the items then you could just use a standard category widget and one click would take you straight into that list in Fen.

> This creates an issue on my setup as every visible item will scroll at the same time when I first enter the screen, which I find distracting. Can you: 1) Make the scrolling optional; or 2) Only scroll the highlighted item?

I'll consider making it optional or removing scrolling completely. 

> Is it possible to add customization options to the Live TV section? For instance: We would like the option to right click and remove items from the 'Recently played channels' section.

I'm not sure if this possible because I believe that the PVR widgets are built in for Kodi. Something you could look into is having a separate Kodi profile for each family member that uses the TV. I don't know much about profiles though as I've never used them.

> We would also like to add widgets for things like our favorite lists or custom groups.

At this time I'm not planning on adding the ability to add and remove widgets from the Kodi PVR section.

> We use IPTV Merge. I'm not sure if any of the items/sections we see come from this addon rather than from default Kodi, but it would be nice to be able to customize.

If you didn't configure an IPTV client inside Kodi, then the Live TV section of the skin shouldn't be showing up on your home screen. 

I sincerely appreciate your detailed feedback and valuable suggestions.

**Comment by Polgara01** [comment]
>Firstly, let me express my gratitude for your kind words and appreciation for the Kodi skin I've built and maintained. It's truly rewarding to know that it has finally provided a satisfactory option for your setup after a decade of using Kodi.

I'm the author of a community project (different community) and I understand how important this is!

> I'm not too sure what you mean by this. You can set the View to either InfoWall or WideInfoWall inside the Fen addon in order to have the same InfoPanel from the widgets.

Let me see if I can explain it without the need for screenshots:  I have "Trending Recent" as a Poster widget.  It is set to allow 20 items so it loads quickly.  At the end of the list it has an arrow for the next page.  Clicking the arrow takes me into Fen.  I understand I can set the view type in Fen, however none of the views match your Poster/InfoPanel setup.  InfoWall is close, but I was looking for horizontal scrolling so it was consistent with the widget.

After reading your responses it occurred to me that it would be better, from a usability perspective, to be able to scroll through multiple pages without leaving the main screen.  Instead of going into Fen, when we click the arrow at the end of the row, would it be possible to load the set of results?  That would be an elegant solution.

> What do you mean 'configure by section'? And are you referring to the the category widget inside a stacked widget? If you are then no, there's no way that I know of to dynamically change the type of widget that displays once it is set up. I will look into this though as it could be a useful feature. Something that you can do as a workaround is make Shortcut folders within Fen and point the category widget to those Shortcut folders. This way you can choose every single item that shows up in the category widget.

If you setup the default Fen movies or TV shows sections as stacked category widgets you have different types of content underneath.  While I prefer Poster for things like "Trending", Poster doesn't work for sections like "Genre" or "Year".  While it would be nice to have that option, I went back to using a regular category widget and just click through into Fen.

> I understand where you're coming from regarding this. I too didn't like that the widget loads right away as soon as it was focused. So I implemented a delay in loading the widget after it is focused. This way you can quickly browse the category widget without loading unnecessary lists in between.
> 
> What you are proposing though poses a problem because clicking the item acts as a shortcut to the Fen addon. This way you can load way more content than the widget has. Also, to be quite honest, this request defeats the entire purpose of the stacked widgets. If you want to wait to click to load the items then you could just use a standard category widget and one click would take you straight into that list in Fen.

How about letting users set the length delay (between x & y) and leave the default as is?  If you want it to load fast, you can load it fast. If you're like me and want a longer delay, you can make it longer.  

> At this time I'm not planning on adding the ability to add and remove widgets from the Kodi PVR section.

Is there any way I can persuade you by presenting a valid use case? :)

We have a list of IPTV links that can only be accessed directly.  They are listed in categories.  We tested them in a category widget in the 'Shows' section and the setup works perfectly.  We also have a handful of favorites from Plex, our antenna, and other sources, that are in their own folders.  Those were tested as Landscape widgets in the 'Shows' section and they also work great.  They add almost no additional load time as the lists are small.  The issue is they don't 'belong' in shows, they 'belong' in 'Live TV' since they are live content.  Yes, it isn't breaking anything, it just makes more sense for live content to be in the live section!

Hopefully this is enough for you to at least ponder it a bit before ruling it out.

> I sincerely appreciate your detailed feedback and valuable suggestions.

And I sincerely appreciate any hobbyist who spends their valuable time producing tools for other hobbyists!  My project started as a personal addon which I decided to share with someone on a forum and then ended up releasing publicly.  I've been maintaining it for years now and it's been mostly enjoyable and rewarding.  I with you well on your journey!

**Comment by ivarbrandt** [comment]
> After reading your responses it occurred to me that it would be better, from a usability perspective, to be able to scroll through multiple pages without leaving the main screen. Instead of going into Fen, when we click the arrow at the end of the row, would it be possible to load the next set of results? That would be an elegant solution.

That would be pretty cool but I don't have a clue how to implement that. Also, I think that it is more useful and functional to click the Next Page item and go directly into a much larger, continued list in Fen. This is because the views in Fen have scrollbars and if you're using any of the Wall views you can browse more content faster, rather than one by one when using the widgets.

> How about letting users set the length delay (between x & y) and leave the default as is? If you want it to load fast, you can load it fast. If you're like me and want a longer delay, you can make it longer.

Let me think about this. I didn't really think I'd be changing the delay because it was designed strike a balance between not loading the lists in between while navigating the category widget and not taking to long to load the item that has been focused. I can definitely see how this would be beneficial though because users scroll at different paces. This is a solid suggestion. Thank you.

> The issue is they don't 'belong' in shows, they 'belong' in 'Live TV' since they are live content.

I completely understand. Right now there is only logic to add widgets to the Movie and Shows sections on the home menu. I won't be adding the ability to add widgets to the default PVR (LiveTV) section as this has its own set of default widgets already. I could however add more sections and the ability to add those respective widgets to each section on the home screen. Opening up more sections though would typically be used for other addons and this is a problem for me because I specifically designed the skin for use with Fen. I don't want to stray from that philosophy. 

> My project started as a personal addon which I decided to share with someone on a forum and then ended up releasing publicly.

That's so cool. What is the name of your addon? I wanna go check it out! 

>  I wish you well on your journey!

Thank you so much. It's been quite enjoyable and rewarding.

**Comment by ivarbrandt** [comment]
For the next update, I've added a customizable delay for stacked widgets, an optional timer to go along with it, and a toggle for whether or not you want the labels on the LandscapeInfo widgets to autoscroll.

**Comment by Polgara01** [comment]
Very nice!  Looking forward to trying it out.

**Comment by TwilightMercy** [comment]
I support adding only one more custom widget section in addition to movies/shows (with custom name chosen)
Too much more will make the skin not FEN specific
